sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -105,10 +117,38 @@ public boolean tryAdd(final EventData eventData) {
             this.sizeInBytes += size;
         }
 
-        this.events.add(eventData);
+        this.events.add(event);
         return true;
     }
 
+    /**
+     * Method to start and end a "Azure.EventHubs.message" span and add the "DiagnosticId" as a property of the message.
+     *
+     * @param eventData The Event to add tracing span for.
+     * @return the updated event data object.
+     */
+    private EventData traceMessageSpan(EventData eventData) {
+        Optional<Object> eventContextData = eventData.getContext().getData(SPAN_CONTEXT_KEY);
+        if (eventContextData.isPresent()) {
+            // if message has context (in case of retries), don't start a message span or add a new context
+            return eventData;
+        } else {
+            // Starting the span makes the sampling decision (nothing is logged at this time)
+            Context eventSpanContext = tracerProvider.startSpan(eventData.getContext(), ProcessKind.MESSAGE);
+            if (eventSpanContext != null) {
```

Comment:
Is there a scenario where tracing is enabled `eventSpanContext` is null? Seems like an incorrectly written tracer.